### PR TITLE
Add code to ensure low-level quests have blue exclamation marks when …

### PR DESCRIPTION
**Pull request by sluggor**
…using tracker

Ensure low-level quest markers are blue when using map tracker

##### CHANGES PROPOSED:

-  Break apart logic checking for low level quest and if quest is repeatable

###### ISSUES ADDRESSED:

Closes https://github.com/azerothcore/azerothcore-wotlk/issues/1862

##### TESTS PERFORMED:

No errors on build. Ran through debugger to make sure variables were being set correctly.

##### HOW TO TEST THE CHANGES:

Bring 80 level character to Isle of Quel'Danas and use "Low Level Quests" tracker on mini map to see all the dailies. Without the fix, they will be yellow. With the fix, they will be blue - the correct color for dailies.

##### KNOWN ISSUES AND TODO LIST:

None

##### Target branch(es):

Master

### CODE REVIEW

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -16807,19 +16807,29 @@ QuestGiverStatus Player::GetQuestDialogStatus(Object* questgiver)
...
-                if (SatisfyQuestLevel(quest, false))
-                {
-                    if (quest->IsAutoComplete())
-                        result2 = DIALOG_STATUS_REWARD_REP;
-                    else if (getLevel() <= (GetQuestLevel(quest) + sWorld->getIntConfig(CONFIG_QUEST_LOW_LEVEL_HIDE_DIFF)))
-                    {
-                        if (quest->IsDaily())
-                            result2 = DIALOG_STATUS_AVAILABLE_REP;
+                if (SatisfyQuestLevel(quest, false)) {
+                    bool bLowLevel = (getLevel() > (GetQuestLevel(quest) + sWorld->getIntConfig(CONFIG_QUEST_LOW_LEVEL_HIDE_DIFF)));
```
Comment (Sdyess) [inline src/server/game/Entities/Player/Player.cpp]: I'm not sure if this is intended to be hungarian notation or supposed to sound like "below level". Either way, I would recommend changing it.
```suggestion
                    bool isBelowLevel = (getLevel() > (GetQuestLevel(quest) + sWorld->getIntConfig(CONFIG_QUEST_LOW_LEVEL_HIDE_DIFF)));
```

Comment (sluggor) [reply src/server/game/Entities/Player/Player.cpp]: It's hungarian notation (boolean), but it's just a habit, we can make it whatever conforms most to the project's coding standards.

Although what do you mean by "changing its uses as well"? It's just a flag to indicate if it's a low level quest or not.

Comment (Sdyess) [reply src/server/game/Entities/Player/Player.cpp]: Sorry for not being more clear, by changing it's uses I just meant to also rename where it's being used. My suggested change had only been where it was declared.

Comment (sluggor) [reply src/server/game/Entities/Player/Player.cpp]: I see. I'm fine either way, but I think "isBelowLevel" is a little misleading. A low level quest, by definition, is below your level, it's just a question of how much.

Once you pass the threshold defined in the worldserver.conf file, the quest is considered "low level" and will no longer show up in game or on the mini map unless explicitly tracked. The relevant configuration options are as follows...

```
#
#    Quests.LowLevelHideDiff
#        Description: Level difference between player and quest level at which quests are
#                     considered low-level and are not shown via exclamation mark (!) at quest
#                     givers.
#        Default:     4  - (Enabled, Hide quests that have 4 levels less than the character)
#                     -1 - (Disabled, Show all available quest marks)

Quests.LowLevelHideDiff = 4

#
#    Quests.HighLevelHideDiff
#        Description: Level difference between player and quest level at which quests are
#                     considered high-level and are not shown via exclamation mark (!) at quest
#                     givers.
#        Default:     7  - (Enabled, Hide quests that have 7 levels more than the character)
#                     -1 - (Disabled, Show all available quest marks)

Quests.HighLevelHideDiff = 7
```

So, once the character's level is higher than the level of the quest plus the value defined by `Quests.LowLevelHideDiff `, it is considered "low level".

As far as naming convention goes, I'm fine with `isLowLevel`, but `isBelowLevel` seems to miss the mark.

Comment (Sdyess) [reply src/server/game/Entities/Player/Player.cpp]: The name you suggest is fine with me as well. Thanks for the pull request 😃

Comment (sluggor) [reply src/server/game/Entities/Player/Player.cpp]: OK, the change is in place. Still pretty new to GitHub, though, so please let me know if anything is off, thanks.

**Comment by wizzymore** [comment]
Tested and it works! Good job

**Comment by BarbzYHOOL** [comment]
Can you run the bash script in your repository root? it's called something like .git_commit...bash

Otherwise, awesome and thank you for contrib! @sluggor

File: src/server/game/Entities/Player/Player.cpp (hunk 1 above)
Comment (BarbzYHOOL) [reply src/server/game/Entities/Player/Player.cpp]: ty sdyess

**Comment by sluggor** [comment]
I don't see any bash scripts like that... What does it do, maybe I can just use git to do it manually?

**Comment by BarbzYHOOL** [comment]
@sluggor it's here in fact, https://github.com/azerothcore/azerothcore-wotlk/tree/master/apps/git_tools

I put it there in case we have other git_tools in the future but I don't know if that was a good idea

**Comment by sluggor** [comment]
Thanks, found it. The commit template has been updated.

**Pull request merged**